### PR TITLE
SQL: Improve test robustness against timezone offset changes

### LINKED ISSUE #68018: JdbcResultSetIT testGettingBooleanValues failing
I've seen 6 failures today.

**Build scan**:
https://gradle-enterprise.elastic.co/s/dpmvt2wywytia/tests/:x-pack:plugin:sql:qa:jdbc:single-node:v7.11.0%23bwcTest/org.elasticsearch.xpack.sql.qa.jdbc.single_node.JdbcResultSetIT/testGettingBooleanValues#1

**Repro line**:

`./gradlew ':x-pack:plugin:sql:qa:jdbc:single-node:v7.11.0#bwcTest' -Dtests.class="org.elasticsearch.xpack.sql.qa.jdbc.single_node.JdbcResultSetIT" -Dtests.method="testGettingBooleanValues" -Dtests.seed=59AC45162D7E3DF1 -Dtests.security.manager=true -Dtests.bwc=true -Dtests.locale=es-GT -Dtests.timezone=Africa/Maseru -Druntime.java=11`

**Reproduces locally?**:
Did not repro lcoally (although I got a random local failure once). I suspect a timing issue.

**Applicable branches**:
Only `master` so far.

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=America/Los_Angeles&tests.container=org.elasticsearch.xpack.sql.qa.jdbc.single_node.JdbcResultSetIT&tests.disabledDistributions=WyJvdXRjb21lOnBhc3NlZCJd&tests.sortField=FAILED&tests.test=testGettingBooleanValues&tests.unstableOnly=true

**Failure excerpt**:
```
org.elasticsearch.xpack.sql.qa.jdbc.single_node.JdbcResultSetIT > testGettingBooleanValues FAILED
    org.junit.ComparisonFailure: expected:<...6-12-17T05:16:37.860[221568]+11:00] of type [DAT...> but was:<...6-12-17T05:16:37.860[]+11:00] of type [DAT...>
        at __randomizedtesting.SeedInfo.seed([59AC45162D7E3DF1:139A13CCAFD7344C]:0)
        at org.junit.Assert.assertEquals(Assert.java:115)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.elasticsearch.xpack.sql.qa.jdbc.ResultSetTestCase.lambda$testGettingBooleanValues$111(ResultSetTestCase.java:930)
        at org.elasticsearch.xpack.sql.qa.jdbc.ResultSetTestCase.doWithQuery(ResultSetTestCase.java:1863)
        at org.elasticsearch.xpack.sql.qa.jdbc.ResultSetTestCase.doWithQuery(ResultSetTestCase.java:1851)
        at org.elasticsearch.xpack.sql.qa.jdbc.ResultSetTestCase.testGettingBooleanValues(ResultSetTestCase.java:917)
```

### PR DESCRIPTION
**Pull request by palesz**
The offset of a timezone is not fixed, can change over time. Countries
can decide to move between offsets/timezones over time. Different
JDK versions have different versions of the timezone -> offset
mapping leading to same instant represented differently in outputs
and errors messages.

This commit improves the error message validation in
the `JdbcResultSetIT` testcases by not asserting the string
representation of the dates in the error message, rather asserting if
they represent the expected instant (epochMillis).

Fixes #68018

### CODE REVIEW

File: x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java [hunk 1]
```diff
@@ -2059,6 +2037,12 @@ private void assertThrowsWritesUnsupportedForUpdate(ThrowingRunnable r) {
         assertThrowsUnsupportedAndExpectErrorMessage(r, "Writes not supported");
     }
 
+    private void assertMessageWithInstant(String actualMessage, Pattern expectedPattern, long epochMillis) {
```
Comment (matriv) [inline x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java:2040]: maybe `assertErrorMessageForDateTimeValues` ?

File: x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java [hunk 2]
```diff
@@ -1157,7 +1135,7 @@ public void testGettingBooleanValues() throws IOException, SQLException {
             assertEquals(false, results.getObject("test_keyword", Boolean.class));
         });
     }
-
+
```
Comment (matriv) [inline x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java:1138]: minor: revert whitespaces

**Review by matriv — APPROVED** [review]
LGTM. Cool approach!

Maybe you could extract the pattern compilation to a method where you pass the target data type needed for the message.

**Review by astefan — APPROVED** [review]
LGTM
Personal opinion: I think it's too much to use a regular expression matcher to basically check equality of two parts of the same string and to extract whatever sits between them. But it gets the job done.

**Review by bpintea — APPROVED** [review]
LGTM

**Review by costin — CHANGES_REQUESTED** [review]
Nice. I would move though the message creation inside the assertion method and clean it up a bit:
```
assertExceptionForInstant(Exception ex, long epoch, Class<?> expectedType)
```

**Review by matriv — APPROVED** [review]
LGTM! thx!

**Review by costin — APPROVED** [review]
LGTM

**Pull request merged**